databricks/koalas/series.py

```diff
@@ -1237,6 +1237,50 @@ def to_list(self):
 
     tolist = to_list
 
+    def drop_duplicates(self, inplace=False):
+        """
+        Return koalas Series with duplicate values removed.
+
+        Parameters
+        ----------
+        inplace: bool, default False
+            If True, performs operation inpalce and returns None.
+
+        Returns
+        -------
+        Series
+            Series with deplicates dropped.
+
+        Examples
+        --------
+        Generate a Series with duplicated entries.
+        >>> s = ks.Series(['lama', 'cow', 'lama', 'beetle', 'lama', 'hippo'],
+        ...               name='animal')
+        >>> s
+        0      lama
+        1       cow
+        2      lama
+        3    beetle
+        4      lama
+        5     hippo
+        Name: animal, dtype: object
+
+        >>> s.drop_duplicates()
+        1       cow
+        0      lama
+        5     hippo
+        3    beetle
+        Name: animal, dtype: object
+        """
+        kdf = self.to_frame().drop_duplicates()
+        internal = kdf._internal.copy()
+
+        if inplace:
+            self._internal = kdf._internal
+            self._kdf = kdf
+        else:
+            return Series(internal, anchor=kdf)
```

Comment:
This loses `internal.scol`, e.g.:

```
>>> s = ks.Series([1, 1, 2, 4, 3])
>>> s.drop_duplicates()
0    1
4    3
2    2
3    4
Name: 0, dtype: int64
>>> (s.drop_duplicates() + 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ueshin/workspace/databricks-koalas/master/databricks/koalas/base.py", line 143, in __add__
    return _column_op(spark.Column.__add__)(self, other)
  File "/Users/ueshin/workspace/databricks-koalas/master/databricks/koalas/base.py", line 57, in wrapper
    scol = f(self._scol, *args)
  File "/Users/ueshin/workspace/databricks-koalas/miniconda/envs/databricks-koalas_3.6/lib/python3.6/site-packages/pyspark/sql/column.py", line 115, in _
    njc = getattr(self._jc, name)(jc)
AttributeError: 'NoneType' object has no attribute '_jc'
```

Reply:
thanks a lot, i think i misused this after reading Series definition and implementation. Please take a look if new change is okay.